apps/cowswap-frontend/src/common/updaters/orders/utils.ts

```diff
@@ -13,73 +17,103 @@ import { getOrder } from 'api/cowProtocol'
+      return {
+        inputAmount: bridgeOrderFromStore.quoteAmounts.swapSellAmount,
+        outputAmount: CurrencyAmount.fromRawAmount(
+          bridgeOrderFromStore.quoteAmounts.bridgeMinReceiveAmount.currency,
+          bridgeOrderFromApi.bridgingParams.outputAmount.toString(),
+        ),
+      }
+    }
+
+    return {
+      inputAmount: bridgeOrderFromStore.quoteAmounts.swapSellAmount,
+      outputAmount: bridgeOrderFromStore.quoteAmounts.bridgeMinReceiveAmount,
+    }
+  }
+
+  // Executed swap order
+  if (isFulfilled) {
+    return {
+      inputAmount: stringToCurrency(executedSellAmount, inputToken),
+      outputAmount: stringToCurrency(executedBuyAmount, outputToken),
+    }
+  }
 
-  let summary: string | undefined = undefined
+  const sellAmount = genericOrder.sellAmount
+  const feeAmount = genericOrder.feeAmount
+  const buyAmount = genericOrder.buyAmount
 
-  if (orderFromStore) {
-    const { inputToken, outputToken, status, kind } = orderFromStore
-    const isFulfilled = status === OrderStatus.FULFILLED
+  // Any other swap orders
+  return {
+    inputAmount: stringToCurrency(sellAmount, inputToken).add(stringToCurrency(feeAmount, inputToken)),
+    outputAmount: stringToCurrency(buyAmount, outputToken),
+  }
+}
 
-    if (!inputToken || !outputToken) return undefined
+function getOrderSummary(genericOrder: EnrichedOrder | Order, { inputAmount, outputAmount }: TradeAmounts): string {
+  const { status, kind } = genericOrder
 
-    // don't show amounts in atoms
-    const inputAmount = isFulfilled
-      ? stringToCurrency(executedSellAmount, inputToken)
-      : // sellAmount doesn't include the fee, so we add it back to not show a different value when the order is traded
-        stringToCurrency(sellAmount, inputToken).add(stringToCurrency(feeAmount, inputToken))
-    const outputAmount = stringToCurrency(isFulfilled ? executedBuyAmount : buyAmount, outputToken)
+  const isFulfilled = status === OrderStatus.FULFILLED
 
+  if (isFulfilled) {
+    return `${stringifyAmount(inputAmount)} ` + t`for` + ` ${stringifyAmount(outputAmount)}`
+  } else {
     const isSell = isSellOrder(kind)
 
-    const inputPrefix = !isFulfilled && !isSell ? t`at most` + ` ` : ''
-    const outputPrefix = !isFulfilled && isSell ? t`at least` + ` ` : ''
+    const inputPrefix = !isSell ? t`at most` + ` ` : ''
+    const outputPrefix = isSell ? t`at least` + ` ` : ''
 
-    summary =
-      `${orderTitle} ${inputPrefix}${formatTokenAmount(inputAmount)} ${formatSymbol(inputAmount.currency.symbol)} ` +
-      t`for` +
-      ` ${outputPrefix}${formatTokenAmount(outputAmount)} ${formatSymbol(outputAmount.currency.symbol)}`
-  } else {
-    // We only have the API order info, let's at least use that
-    summary = `${orderTitle} ${sellToken} ` + t`for` + ` ${buyToken}`
+    return (
+      `${inputPrefix}${stringifyAmount(inputAmount)} ` + t`for` + ` ${outputPrefix}${stringifyAmount(outputAmount)}`
```

Comment:
I think it's not the best way to handle i18n, it could lead to loosing the context

I suggest to use t for the whole sentence:
    const inputDisplay = stringifyAmount(inputAmount)
    const outputDisplay = stringifyAmount(outputAmount)

    return t`${inputPrefix}${inputDisplay} for ${outputPrefix}${outputDisplay}`

Reply:
100% agree.
I've deleted this function in the next PR: https://github.com/cowprotocol/cowswap/pull/6773